required/offline_API_wrapper.js

```diff
@@ -68,7 +81,18 @@ var API = {
 	LMSStore: function(force) {
 		if (window.ISCOOKIELMS === false) return;
 		if (!force && API.cookie("_spoor") === undefined) return;
-		API.cookie("_spoor", JSON.stringify(this.data));
+
+		var stringified = JSON.stringify(this.data);
+
+		API.cookie("_spoor", stringified);
+
+		// a length mismatch will most likely indicate cookie storage limit exceeded
+		if (API.cookie("_spoor").length != stringified.length) {
```

Comment:
You're not wrong, @oliverfoster! Something like this?

```js
function storageWarning() {
	this.__storageWarningTimeoutId = null;
	var Adapt = require ? require('coreJS/adapt') : null;
	if (!Adapt) return;
	var spoor = Adapt.config ? Adapt.config.get('_spoor') : null;
	if (!spoor) return;
	var settings = config._advancedSettings;
	var notificationMethod = alert;
	if (settings && settings._suppressErrors) notificationMethod = console.error;
	notificationMethod('Warning: possible cookie storage limit exceeded - tracking may malfunction');
}
```

The limited-whitespace thing feels weird.